room_assistance/indico_room_assistance/forms.py

```diff
@@ -39,12 +39,24 @@ def process_formdata(self, valuelist):
         dts = []
         tzinfo = pytz.timezone(config.DEFAULT_TIMEZONE)
         for req_date, req_time in self.data.iteritems():
-            dt = tzinfo.localize(dateutil.parser.parse('{} {}'.format(req_date, req_time)))
+            if not req_time:
+                dt = datetime.strptime(req_date, '%Y-%m-%d').date()
+            else:
+                dt = tzinfo.localize(dateutil.parser.parse('{} {}'.format(req_date, req_time)))
             dts.append(dt)
         self.data = dts
 
     def _value(self):
-        return {dt.date().isoformat(): dt.strftime('%H:%M') for dt in self.data} if self.data else {}
+        if not self.data:
+            return {}
+
+        data = {}
+        for dt in self.data:
+            if type(dt) is datetime:
```

Comment:
`isinstance()`?

Reply:
I know but I wanted to be clear here since `isinstance(datetime.now(), datetime.date)` returns True.

Reply:
WTF, why is a datetime a date subclass (but not a time subclass)?!